lib/screens/home/home_repository.dart

```diff
@@ -0,0 +1,75 @@
+import 'dart:convert';
+import 'dart:math';
+
+import 'package:ecellapp/core/res/errors.dart';
+import 'package:ecellapp/models/user.dart';
+import 'package:flutter/material.dart';
+import 'package:ecellapp/core/res/strings.dart';
+import 'package:ecellapp/core/utils/injection.dart';
+import 'package:ecellapp/core/utils/logger.dart';
+import 'package:http/http.dart' as http;
+import 'package:shared_preferences/shared_preferences.dart';
+
+@immutable
+abstract class HomeRepository {
+  /// Takes in `token` , gives the user details and throws a suitable exception if something goes wrong.
+  Future<User> profile(String token);
```

Comment:
🔴 
1. function name should always be verb.
2. the `token` argument is not used in either fake or api classes. remove it.

Reply:
done